Gems/ONNX/Code/Source/Clients/Mnist.cpp

```diff
@@ -0,0 +1,168 @@
+        // i.e. a pure black pixel is a 0, anything else is a 1.
+        // Bear in mind that the images in the dataset are flipped compared to how we'd usually think,
+        // so the background is black and the actual digit is white.
+        for (int y = 0; y < m_imageHeight; y++)
+        {
+            for (int x = 0; x < m_imageWidth; x++)
+            {
+                int content = static_cast<int>(buffer[(y)*m_imageWidth + x]);
+                if (content == 0)
+                {
+                    m_input[0][m_imageWidth * y + x] = 0.0f;
+                }
+                else
+                {
+                    m_input[0][m_imageHeight * y + x] = 1.0f;
+                }
+            }
+        }
+    }
+
+    MnistReturnValues MnistExample(Mnist& mnist, const char* path)
+    {
+        mnist.LoadImage(path);
+        mnist.Run(mnist.m_input);
+        mnist.GetResult();
+
+        MnistReturnValues returnValues;
+        returnValues.m_inference = mnist.m_result;
+        returnValues.m_runtime = mnist.m_delta;
+        return (returnValues);
+    }
+
+    InferenceData RunMnistSuite(int testsPerDigit, bool cudaEnable)
+    {
+        // Initialises and loads the mnist model.
+        // The same instance of the model is used for all runs.
+        Mnist mnist;
+        AZStd::vector<AZStd::vector<float>> input(0);
+        AZStd::vector<float> image(mnist.m_imageSize);
+        input.push_back(image);
+        mnist.m_input = input;
+        //AZStd::vector<float> output(10);
+        //mnist.m_output = output;
+
+        Mnist::InitSettings modelInitSettings;
+
+        if (cudaEnable)
+        {
+            modelInitSettings.m_modelName = "MNIST CUDA (Precomputed)";
+            modelInitSettings.m_modelColor = AZ::Color::CreateFromRgba(56, 229, 59, 255);
+            modelInitSettings.m_cudaEnable = true;
+        }
+        else
+        {
+            modelInitSettings.m_modelName = "MNIST (Precomputed)";
+        }
+
+        mnist.Load(modelInitSettings);
+
+        int numOfEach = testsPerDigit;
+        int totalFiles = 0;
+        int64_t numOfCorrectInferences = 0;
+        float totalRuntimeInMilliseconds = 0;
+        AZ::IO::FixedMaxPath mnistTestImageRoot;
+
+        // This bit cycles through the folder with all the mnist test images, calling MnistExample() for the specified number of each digit.
+        // The structure of the folder is as such: /testing/{digit}/{random_integer}.png    e.g /testing/3/10.png
+        auto TestImage = [&mnist, &numOfCorrectInferences, &totalFiles, &totalRuntimeInMilliseconds, &mnistTestImageRoot, numOfEach](AZ::IO::Path digitFilePath, bool isFile) -> bool
+        {
+            if (!isFile)
+            {
+                AZ::IO::FixedMaxPath directoryName = digitFilePath.Filename();
+                char* directoryEnd;
+                AZ::s64 digit = strtoll(directoryName.c_str(), &directoryEnd, 10);
```

Comment:
@amzn-mike I believe you are aliasing the C++17 https://en.cppreference.com/w/cpp/utility/from_chars into the AZStd:: namespace correct?